drivers/spi/spi_mcux_lpspi.c

```diff
@@ -773,87 +790,54 @@ static void spi_mcux_iodev_start(const struct device *dev)
 		break;
 	default:
 		LOG_ERR("Invalid op code %d for submission %p\n", sqe->op, (void *)sqe);
-
-		spi_mcux_iodev_next(dev, true);
-		rtio_iodev_sqe_err(txn_head, -EINVAL);
-		spi_mcux_iodev_complete(dev, 0);
+		spi_mcux_iodev_complete(dev, -EINVAL);
 		return;
 	}
 
 	data->transfer_len = transfer.dataSize;
 
-	k_spinlock_key_t key = spi_spin_lock(dev);
+	k_spinlock_key_t key = spi_spin_lock(rtio_ctx);
```

Comment:
It's unclear why this spinlock is needed here off hand, take a close look at this if you don't mind. If there's a reason you can see add a comment. If no clarity this then this is something we should look a little harder at.

This seems to be the sole reason to expose the spin lock, which otherwise should be hidden to the drivers in the spi_rtio context.

Reply:
Looked a bit deeper and I don't believe is required. I'll scrub this and make the spin lock functions private